tdrs-backend/tdpservice/reports/models.py

```diff
@@ -1,10 +1,31 @@
 """Define report models."""
+import os
 
+from django.conf import settings
 from django.db import models
 from django.db.models import Max
+from storages.backends.s3boto3 import S3Boto3Storage
 
-from ..stts.models import STT
-from ..users.models import User
+from tdpservice.stts.models import STT
+from tdpservice.users.models import User
+
+
+def get_s3_upload_path(instance, filename):
+    """Produce a unique upload path for S3 files for a given STT and Quarter."""
+    return os.path.join(
+        f'data_files/{instance.stt.id}/{instance.quarter}',
```

Comment:
After this is merged we will have a follow on issue (TODO: Make and link issue) to fully implement the built in S3 versioning of files. This will negate the need to generate and store a slug for each ReportFile.